plugins/catalog-backend/config.d.ts

```diff
@@ -42,6 +44,20 @@ export interface Config {
       allow: Array<string>;
     }>;
 
+    /**
+     * Mode defines the overall behaviour mode of the catalog.
+     *
+     * Setting the mode to 'readwrite' you allow users to register their own
+     * components. This is the default value.
+     *
+     * Setting the mode to 'readonly' configures catalog to only allow reads.
+     * This can be used in combination with static locations to only serve
+     * operator provided locations. Effectively this removes the ability to
+     * register new components to a running backstage instance.
+     *
+     */
+    mode?: Mode;
```

Comment:
Let's avoid the separate type definition, it shares namespace with all other config schema definitions. Just `mode?: 'readonly' | 'readwrite'` instead.

Reply:
It was to allow us to set the type in the handler to use a static type: https://github.com/backstage/backstage/pull/5034/files#diff-a63ba4ab9ace8b05bf9b886d1f81f808a8e20ad58799be258e38ea9f0f708fe0R68-R69

A typo in the default value would not get catched and we couldn't force the type in `requireReadWriteMode` function as well.

I'm a bit unsure what the general way of doing this is in backstage so please enlighten me 😉  I would love for our code to be as strict as possible in regards to these configurations. 😄

Reply:
By switching to use `readonly: bool` this is "solved" :)